benchmark/run.js

```diff
@@ -10,31 +10,56 @@ const cli = CLI(`usage: ./node run.js [options] [--] <category> ...
 
   --filter pattern          string to filter benchmark scripts
   --set    variable=value   set benchmark variable (can be repeated)
+  --format [simple|csv]      optional value that specifies the output format
 `, {
   arrayArgs: ['set']
 });
 const benchmarks = cli.benchmarks();
+const validFormats = ['csv', 'simple'];
 
 if (benchmarks.length === 0) {
-  console.error('no benchmarks found');
-  process.exit(1);
+  console.error('No benchmarks found');
+  // node documentation suggests using this exit method when you need to still
+  // output data
+  process.exitCode = 1;
+  return;
 }
 
+const format = cli.optional.format || 'simple';
+if (!validFormats.includes(cli.optional.format)) {
+  console.error('Invalid format detected');
+  // node documentation suggests using this exit method when you need to still
+  // output data
+  process.exitCode = 1;
+  return;
+}
+
+
 (function recursive(i) {
   const filename = benchmarks[i];
   const child = fork(path.resolve(__dirname, filename), cli.optional.set);
 
   console.log();
-  console.log(filename);
+  if (format == 'csv') {
+    console.log('"filename","configuration","rate","time"');
```

Comment:
This is going to output a csv header for every file. A csv header should only appear once.